pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -104,8 +104,8 @@ public void process()
       // Update single-column indices, like inverted index, json index etc.
       IndexCreatorProvider indexCreatorProvider = IndexingOverrides.getIndexCreatorProvider();
       for (ColumnIndexType type : ColumnIndexType.values()) {
-        IndexHandlerFactory.getIndexHandler(type, _segmentMetadata, _indexLoadingConfig)
-            .updateIndices(segmentWriter, indexCreatorProvider);
+        IndexHandler handler = IndexHandlerFactory.getIndexHandler(type, _segmentMetadata, _indexLoadingConfig);
+        handler.updateIndices(segmentWriter, indexCreatorProvider);
```

Comment:
just curious, is there a need to handle forward index handler first before calling other index handlers? I think range index will need to be rewritten based on the compression codec change, right?

Reply:
Good point. It is required that we process forwardIndexHandler changes first. Let me add a comment in ColumnIndexType to clarify this.

One thing to note is that, range index need not be rewritten for compression code change. Range index only needs to be rewritten if dictEnabled changes.

Reply:
Oops, yes sorry that's what I meant about the range index.